checker.c

```diff
@@ -1,21 +1,164 @@
+ ***
+ ***     \par         File_description
+ ***                  This file explores more about functions
+ ***
+ *****************************************************************************/
+ 
+/*=============================================================================
+ =======                            INCLUDES                             =======
+ ============================================================================*/
+ /*------ standard includes -------*/
 #include <stdio.h>
 #include <assert.h>
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+/*------ module includes --------*/
+#include "checker.h"
+
+/*=============================================================================
+ =======               DEFINES & MACROS FOR GENERAL PURPOSE              =======
+ =============================================================================*/
+
+/*=============================================================================
+ =======                       CONSTANTS  &  TYPES                       =======
+ =============================================================================*/
+
+/*=============================================================================
+ =======                VARIABLES & MESSAGES & RESSOURCEN                =======
+ ==============================================================================*/
+static BMS_parameters_s BMS_parameters_attributes = {
+		FALSE,
+		FALSE,
+		FALSE,
+		FALSE
+};
+/*=============================================================================
+ =======                              METHODS                            =======
+ =============================================================================*/
+
+/*---------------------------------------------------------------------------*/
+/*     FUNCTION: BMS_checkAttribute_OutOfRange
+ */
+/*!    \brief   Function to check the range of parameters
+ *
+ *
+ *     \param    	bms_attribute -> the parameter for which range needs to be checked
+					attribute_Min_Value -> Lower range of the parameter
+					attribute_Max_Value -> Upper range of the parameter
+					param[]	-> Parameter name to be printed
+ *     \returns  	void
+ *
+ *//*------------------------------------------------------------------------*/
+bool BMS_checkAttribute_OutOfRange(float bms_attribute, float attribute_Min_Value, float attribute_Max_Value, char param[])
+{
+	bool retAtributeStatus = FALSE;
+	if(bms_attribute < attribute_Min_Value || bms_attribute > attribute_Max_Value) 
+	{
+		BMS_printParameterStatus(param,"out of range!");
+		retAtributeStatus = TRUE;
+		BMS_setBMSStatus(TRUE);									
```

Comment:
Good accumulation of status without increasing complexity, even when more parameters are added!